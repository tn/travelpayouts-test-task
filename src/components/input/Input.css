@import '../../Config.css';

.field {
    min-width: 300px;
    height: 50px;
    margin: 0;
    padding: 0 17px;
    font-family: var(--baseFontFamily);
    font-size: 15px;
    border: 1px solid var(--defaultButtonBorder);
    border-radius: 4px;
    outline: 0;
}

.field:focus {
    border-color: var(--fieldActiveColor);
    box-shadow: 0 0 3px var(--fieldActiveColor);
}

.field__label {
    display: block;
    margin: 0 0 7px;
    font-size: 11px;
    line-height: 16px;
    color: var(--secondaryTextColor);
    text-transform: uppercase;
}
