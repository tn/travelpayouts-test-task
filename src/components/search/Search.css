.search {
    margin: 0 0 24px;
}

.search .field {
    margin-right: 8px;
}
