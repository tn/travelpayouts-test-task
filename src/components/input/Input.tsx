import * as React from 'react'
import './Input.css'

interface IInputProps {
  labelId: string
  value?: string
  readOnly?: boolean
  onChange?: React.ChangeEventHandler
  autoComplete?: string
}

const Input: React.SFC<IInputProps> = ({ children, labelId, ...rest }) => (
  <React.Fragment>
    <label className='field__label' htmlFor={labelId}>{children}</label>
    <input className='field' id={labelId} {...rest} />
  </React.Fragment>
)

export default Input
