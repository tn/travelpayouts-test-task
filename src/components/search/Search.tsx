import * as React from 'react'
import './Search.css'
import Button from '../button/Button'
import Input from '../input/Input'
import LangContext from '../../Context'

interface ISearchProps {
  onSearch: Function
}

type InputEvent = React.ChangeEvent<HTMLInputElement>
type FormEvent = React.FormEvent<HTMLFormElement>

const Search: React.SFC<ISearchProps> = (props) => {
  const l = React.useContext(LangContext)
  const onChange = (event: InputEvent) => {
    props.onSearch(event.currentTarget.value)
  }

  const onReset = (event: FormEvent) => {
    props.onSearch('')
  }

  return (
    <form className='search' onReset={onReset} autoComplete='off'>
      <Input onChange={onChange} type='text' labelId='search' autoComplete='off'>{l('filter')}</Input>
      <Button component='input' type='reset' value={l('reset')} />
    </form>
  )
}

export default Search
