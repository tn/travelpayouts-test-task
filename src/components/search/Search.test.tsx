import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Search from './Search'
import Input from '../input/Input'

configure({ adapter: new Adapter() })

describe('Search', () => {
  const props = {
    onSearch: jest.fn()
  }

  it('should be rendered', () => {
    const wrapper = renderer.create(<Search {...props} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should trigger callback', () => {
    const wrapper = shallow(<Search {...props} />)

    wrapper.find(Input).simulate('change', { currentTarget: { value: '1' } })

    expect(props.onSearch).toHaveBeenCalledTimes(1)
  })
})
