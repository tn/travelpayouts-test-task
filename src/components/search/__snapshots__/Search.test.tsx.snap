// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should be rendered 1`] = `
<form
  autoComplete="off"
  className="search"
  onReset={[Function]}
>
  <label
    className="field__label"
    htmlFor="search"
  >
    Filter
  </label>
  <input
    autoComplete="off"
    className="field"
    id="search"
    onChange={[Function]}
    type="text"
  />
  <input
    className="button "
    type="reset"
    value="Reset"
  />
</form>
`;
